pocs/filterwheel/zwo.py

```diff
@@ -0,0 +1,173 @@
+            'device name' (e.g. 'EFW_7_0') can be used to select the desired one. See docstring
+            of `pocs.filterwheel.libefw.EFWDriver.get_devices()` for details.
+        initial_filter (str or int): Name of filter (or integer position) to move to when
+            initialising filter.
+    """
+
+    _driver = None
+    _filter_wheels = {}
+    _assigned_filterwheels = set()
+
+    def __init__(self,
+                 name='ZWO Filter Wheel',
+                 model='zwo',
+                 camera=None,
+                 filter_names=None,
+                 timeout=10 * u.second,
+                 serial_number=None,
+                 library_path=None,
+                 unidirectional=True,
+                 device_name=None,
+                 initial_filter=None,
+                 *args, **kwargs):
+        if camera and not isinstance(camera, AbstractCamera):
+            msg = f"Camera must be an instance of pocs.camera.camera.AbstractCamera, got {camera}."
+            self.logger.error(msg)
+            raise ValueError(msg)
+        super().__init__(name=name,
+                         model=model,
+                         camera=camera,
+                         filter_names=filter_names,
+                         timeout=timeout,
+                         serial_number=serial_number,
+                         *args, **kwargs)
+
+        if FilterWheel._driver is None:
+            # Initialise the driver if it hasn't already been done
+            FilterWheel._driver = EFWDriver(library_path=library_path)
+
+        self._device_name = device_name
+        self.connect()
+        self.is_unidirectional = unidirectional
+        if initial_filter:
+            self.move_to(initial_filter, blocking=True)
+
+    def __del__(self):
+        """Attempt some clean up."""
+        with suppress(AttributeError):
+            device_name = self._device_name
+            FilterWheel._assigned_filterwheels.discard(device_name)
+            self.logger.debug('Removed {} from assigned filterwheels list'.format(device_name))
+
+##################################################################################################
+# Properties
+##################################################################################################
+
+    @AbstractFilterWheel.position.getter
+    def position(self):
+        """ Current integer position of the filter wheel """
+        return self._driver.get_position(self._handle) + 1  # 1-based numbering
+
+    @property
+    def is_moving(self):
+        """ Is the filterwheel currently moving """
+        return not self._move_event.is_set()
+
+    @property
+    def is_unidirectional(self):
+        return self._driver.get_direction(self._handle)
+
+    # ZWO filterwheels can be set to be unidirectional or bidirectional.
+    @is_unidirectional.setter
+    def is_unidirectional(self, unidirectional):
+        self._driver.set_direction(self._handle, bool(unidirectional))
```

Comment:
I don't see `set_direction`

Reply:
It's a method of the `pocs.filterwheel.libefw.EFWDriver` class, which `self._driver` is an instance of.